src/pages/iou/request/step/IOURequestStepDistance.tsx

```diff
@@ -90,7 +90,7 @@ function IOURequestStepDistance({
     const previousValidatedWaypoints = usePrevious(validatedWaypoints);
     const haveValidatedWaypointsChanged = !isEqual(previousValidatedWaypoints, validatedWaypoints);
     const isRouteAbsentWithoutErrors = !hasRoute && !hasRouteError;
-    const shouldFetchRoute = (isRouteAbsentWithoutErrors || haveValidatedWaypointsChanged) && !isLoadingRoute && Object.keys(validatedWaypoints).length > 1;
+    const shouldFetchRoute = (!transaction?.routes || isRouteAbsentWithoutErrors || haveValidatedWaypointsChanged) && !isLoadingRoute && Object.keys(validatedWaypoints).length > 1;
```

Comment:
Let's be a bit more precise with this to guard against weird edge cases that might arise. Let's look for `(transaction?.routes?.route0?.coordinates ?? []).length` or something similar.